libtiny_ui/src/lib.rs

```diff
@@ -35,23 +35,29 @@ pub enum MsgSource {
 impl MsgSource {
     pub fn serv_name(&self) -> &str {
         match *self {
-            MsgSource::Serv { ref serv }
+            MsgSource::Serv { ref serv, .. }
             | MsgSource::Chan { ref serv, .. }
             | MsgSource::User { ref serv, .. } => serv,
         }
     }
 
     pub fn to_target(&self) -> MsgTarget {
         match *self {
-            MsgSource::Serv { ref serv } => MsgTarget::Server { serv },
+            MsgSource::Serv { ref serv, .. } => MsgTarget::Server { serv },
             MsgSource::Chan { ref serv, ref chan } => MsgTarget::Chan { serv, chan },
             MsgSource::User { ref serv, ref nick } => MsgTarget::User { serv, nick },
         }
     }
 
     pub fn visible_name(&self) -> &str {
         match *self {
-            MsgSource::Serv { ref serv, .. } => serv,
+            MsgSource::Serv {
+                ref serv,
+                ref alias,
+            } => match alias {
+                Some(alias) => alias,
```

Comment:
all of this for this one call!

Reply:
Hmm.. I wonder if we could add a field to `Tab` instead of `MsgSource`, and update the `Tab::visible_name` method to use that field. We'd only use `MsgSource::visible_name` when the field is `None`. Does this make sense?

Reply:
This sounds way better actually. Will fix it.